### ISSUE TITLE
Way to follow symlink?

### ISSUE THREAD

**Issue by nerdfever** [shundhammer/qdirstat#240]
Is it possible to add a method to follow a symlink manually (a button or right-click option)?

Often I right-click on a symlinked directory on my desktop, choose qdirstat, and see something like:

![image](https://github.com/shundhammer/qdirstat/assets/9833299/755f7f37-74b0-4627-95c3-9cdb0a1ade12)

It would be great if I could click on the link, a button, or something to make qdirstat follow the link and run on the target.

**Comment by shundhammer** [comment]
If it's just for following a symlink passed as a directory to read on the command line (which is what the .desktop file magic actually does), that should be easy enough.

Stay tuned.

**Comment by shundhammer** [comment]
```
[sh @ uruk] ...src/qdirstat/src 64 % file follow* nirvana
followme:  symbolic link to /work/home/sh/src
followme2: symbolic link to followme
nirvana:   broken symbolic link to ../../nowhere

[sh @ uruk] ...src/qdirstat/src 65 % qdirstat followme2
```
![QDirStat-valid-symlink](https://github.com/shundhammer/qdirstat/assets/11538225/b49bed83-4a37-4339-b0f8-1c9af62836f0)

```
[sh @ uruk] ...src/qdirstat/src 66 % qdirstat nirvana  
```
![QDirStat-broken-symlink](https://github.com/shundhammer/qdirstat/assets/11538225/89d4df35-ac90-4d2e-9304-58379dd3a023)

**Comment by shundhammer** [comment]
When I read this issue first, I thought it was about generally following symlinks from a an in-memory directory tree. I had thought about that in the past several times, and some day it will probably come. Here are my thoughts about that - just so they don't get lost:

## Following Symlinks in General

Unfortunately, that's a lot more difficult than it sounds.

A symlink can point to another symlink; that's one issue (and it's quite common). I am not sure if such a function should follow a chain of symlinks all the way to the target, or just one level; if there is a problem in that chain such as no permissions for one of the intermediate directories, it would be impossible to find out. But let's say the normal operation should follow a chain of symlinks to the final target.

Another problem is that QDirStat usually only reads a subtree of a complete filesystem's directory hierarchy; for example `/home/kilroy` if that user starts it. But when that subtree contains a link _outside_ of the tree like `/home/kilroy/project -> /home/common/projects/foo`, then what? It would have to report to the user that this symlink cannot be followed, or offer to read the complete `/home` hierarchy.

And if that destination link is also a link, pointing outside of that other team as well, an even larger tree would have to be read.

Okay, part of this could be alleviated by not using QDirStat's in-memory tree, but reading directly from disk for such calls. But that would open a host of other problems if the in-memory tree and the on-disk tree are no longer in sync (i.e. if the in-memory tree has become outdated).

Another issue is the user interface. Right now, the 'Details' panel on the right already shows a symlink's (direct) target, and when it's a bit longer (e.g. a full path), it's underlined showing that it's clickable. Not sure if that's in the last released version 1.8.1 yet, but it's in the current development version which will become 1.9 soon.

When clicked, right now it shows the full target in a pop-up. But that could be extended; it could be turned into a pop-up menu with the full path on top, and then menu options like "Follow link (first level)" and "Follow link (all levels)".

And then... well, exactly what? Probably navigate in the tree to the target, very much like when you click on a treemap tile. Of course that would throw the user completely out of context (similar to clicking on a treemap tile - where that is expected, though), and it wouldn't be trivial to go back to the previous location.

Yes, we now have a browser-like history buttons ("back" and "forward"), but that goes back just to the previous _directory_, not to the exact item; the symlink in this case. You'd have to scroll down the list to get to the symlink. Not the end of the world, but annoying.
